planner/optimize.go

```diff
@@ -105,23 +105,31 @@ func Optimize(ctx context.Context, sctx sessionctx.Context, node ast.Node, is in
 		sctx.GetSessionVars().StmtCtx.AppendWarning(warn)
 	}
 	warns = warns[:0]
+
+	stmtNode, isStmtNode := node.(ast.StmtNode)
+	useBinding := sessVars.UsePlanBaselines && isStmtNode
+	var bindRecord *bindinfo.BindRecord
+	var scope string
+	if useBinding {
+		bindRecord, scope = getBindRecord(sctx, stmtNode)
+		if bindRecord == nil || len(bindRecord.Bindings) == 0 {
+			useBinding = false
+		}
+	}
+	if isStmtNode {
```

Comment:
Why to add the check for `isStmtNode`?

Reply:
`TryAddExtraLimit` cannot only support `StmtNode`.